### PR TITLE
Feature/auto audience segmentation

### PR DESCRIPTION
**Pull request by zvizdo**
# Description

Added a new pipeline to MAJ - Auto Audience Segmentation.
After the initial MAJ installation, a [notebook](https://github.com/adswerve/marketing-analytics-jumpstart/blob/feature/auto-audience-segmentation/python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest.ipynb) can be accessed which walks the executor through a process generating interest-based clusters. 
Optionally, the resulting model can be deployed to Vertex AI Models.
The model will then be used for a daily cluster/segment prediction with predictions pushed to GA4.

# How has this been tested?

 - Fresh install of MAJ
 - Unit tested the feature engineering pipeline via the existing unit test
 - Created a new test for the prediction pipeline

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [x] I have assigned a reviewer and messaged them

### CODE REVIEW

**Comment by chmstimoteo** [comment]
@zvizdo what is the difference between the notebooks `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest.ipynb` and `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest-direct.ipynb`?

File: sql/procedure/auto_audience_segmentation_inference_preparation.sqlx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  -- distributed under the License is distributed on an "AS IS" BASIS,
+  -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  -- See the License for the specific language governing permissions and
+  -- limitations under the License.
+  -- Setting procedure to lookback from the day before `inference_date`
+
+DECLARE lookback_days INT64 DEFAULT 15;
+SET inference_date = DATE_SUB(inference_date, INTERVAL 1 DAY);
+
+CALL `{{project_id}}.{{dataset}}.create_dataset`(inference_date, inference_date, lookback_days);
```
Comment (chmstimoteo) [inline sql/procedure/auto_audience_segmentation_inference_preparation.sqlx:19]: Please use a more intuitive name for the training table. Something like `audience_segmentation_training_15`

**Comment by zvizdo** [comment]
> @zvizdo what is the difference between the notebooks `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest.ipynb` and `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest-direct.ipynb`?

-direct just doesn't use the presentation layer, but the source GA4 directly. This can be deleted.

File: sql/procedure/auto_audience_segmentation_inference_preparation.sqlx (hunk 1 above)
Comment (zvizdo) [reply sql/procedure/auto_audience_segmentation_inference_preparation.sqlx:19]: The output here generates a table called `v_auto_audience_segmentation_inference_15`, dataset is just an intermediate table that gets deleted at the end of the procedure.

**Comment by chmstimoteo** [comment]
> > @zvizdo what is the difference between the notebooks `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest.ipynb` and `python/notebooks/vertex_ai_auto_audience_segmentation_ga4_interest-direct.ipynb`?
> 
> -direct just doesn't use the presentation layer, but the source GA4 directly. This can be deleted.

Yes, please delete it @zvizdo

**Review by chmstimoteo — APPROVED** [review]
LGTM!

**Pull request merged**